packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m

```diff
@@ -327,6 +341,83 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
   return self;
 }
 
+/**
+ * Sets up the picture in picture controller and assigns the AVPictureInPictureControllerDelegate to
+ * the controller.
+ */
+- (void)setupPiPController {
+  if (@available(macOS 10.15, *)) {
+    if ([AVPictureInPictureController isPictureInPictureSupported]) {
+      self.pictureInPictureController =
+          [[AVPictureInPictureController alloc] initWithPlayerLayer:self.playerLayer];
+      [self setAutomaticallyStartsPictureInPicture:NO];
+      self.pictureInPictureController.delegate = self;
+    }
+  } else {
+    // We don't do anything here because there is no setup required below macOS 10.15.
+  }
+}
+
+- (void)setAutomaticallyStartsPictureInPicture:
+    (BOOL)canStartPictureInPictureAutomaticallyFromInline {
+  if (!self.pictureInPictureController) return;
+#if TARGET_OS_IOS
+  if (@available(iOS 14.2, *)) {
+    self.pictureInPictureController.canStartPictureInPictureAutomaticallyFromInline =
+        canStartPictureInPictureAutomaticallyFromInline;
+  }
+#endif
+}
+
+- (void)setPictureInPictureOverlaySettings:(CGRect)frame {
+  if (_player) {
+    self.playerLayer.frame = frame;
+  }
+}
+
+- (void)startOrStopPictureInPicture:(BOOL)shouldPictureInPictureStart {
```

Comment:
It seems very confusing from an API standpoint that both
```objc
[self startOrStopPictureInPicture:foo];
```

(which is already hard to parse, since the bool argument isn't named by the signature in violation of standard Obj-C naming conventions)

and
```objc
self.isPictureInPictureStarted = foo;
```

exist, and do different things.

Why not just override the setter of the property to do what this method does, and eliminate this duplicate-sounding method?